bash_completion

```diff
@@ -158,10 +158,31 @@ quote_readline()
     printf %s "$ret"
 } # quote_readline()
 
+# shellcheck disable=SC1003
+_comp_dequote__initialize()
+{
+    unset -f "$FUNCNAME"
+    local param='\$([_a-zA-Z][_a-zA-Z0-9]*|[-*@#?$!0-9_])|\$\{[!#]?([_a-zA-Z][_a-zA-Z0-9]*(\[([0-9]+|[*@])\])?|[-*@#?$!0-9_])\}'
+    local quote='\\.|'\''[^'\'']*'\''|"([^\"$`!]|'$param'|\\.)*"|\$'\''([^\'\'']|\\.)*'\'''
+    _comp_dequote__regex_safe_word='^([^\'\''"$`;&|<>()!]|'$quote'|'$param')*$'
+}
+_comp_dequote__initialize
+
+_comp_dequote()
```

Comment:
Can we add a few lines of documentation here, e.g. in what kind of cases one should look into using this, is it a safe replacement for expansion with `eval`, etc?

Reply:
Thank you for the suggestion! I'm sorry that this branch seems to be somehow incomplete because I initially created it two months ago and forgot that I left it alone in an incomplete state. I have added the description b39640b1. Also, as I have described in the added description, I think I need to explain the caveats/decisions of the current implementation.

- In the new implementation, the "*unsafe*" words that contain command substitutions or other arbitrary command execution will not be expanded.
- Nevertheless, the parameter expansions are expanded although it might contain command substitutions through the name references. I think the name references should be generally designed to respect a certain kind of referential transparency even if that involves some command executions, so that shouldn't be the problem as far as with the sane setups.